pandas/io/excel.py

```diff
@@ -708,7 +708,12 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0):
         for cell in cells:
             colletter = get_column_letter(startcol + cell.col + 1)
             xcell = wks.cell("%s%s" % (colletter, startrow + cell.row + 1))
-            xcell.value = _conv_value(cell.val)
+            new_data_type = xcell.data_type_for_value(cell.val)
+            if (isinstance(cell.val, compat.string_types)
+                    and new_data_type != xcell.TYPE_STRING):
```

Comment:
you can do the ``.data_type_for_value`` inside the if (e.g. make another condition), then we ONLY do this if we have a string types